en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md

```diff
@@ -5,34 +5,68 @@
 Given an array of integers containing `n + 1` integers, where each integer is in the range `[1, n]` inclusive. If there is only one duplicate number in the input array, this algorithm returns the duplicate number without modifying the original array, otherwise, it returns -1.
 
 ## Approach
+- We can imagine the array `arr` as a directed graph where each element is a node. Node with value `arr[i]` will point to node with value `arr[arr[i]]`.
+- The problem now becomes finding the node where two nodes with same value point towards.
```

Comment:
That sounds like something where two nodes point to the same node, which is not a cycle. A cycle is when there is a path from a node to itself. So for example in `[1, 0]`, you start at 0, then go to 1, then go back to zero.